site/src/pages/release-notes/1.27.0.mdx

```diff
@@ -0,0 +1,228 @@
+  ```java
+  ClientFactory clientFactory =
+    ClientFactory
+      .builder()
+      .connectionPoolListener(ConnectionPoolListener.metricCollecting(meterRegistry))
+      .build();
+  WebClient
+    .builder("https://my-backend.com")
+    .factory(clientFactory)
+    .build();
+  ```
+- **Graceful Executor Shutdown**: You can now gracefully terminate a custom `ExecutorService` in sync with the
+- <type://Server>'s shutdown procedure using <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)>
+  and <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)> #5031 #5087
+  ```java
+  ServerListener
+    .builder()
+    // Shutdown the executor service when the server is stopping.
+    .shutdownWhenStopping(preStopExecutorService)
+    // Shutdown the executor service when the server is stopped.
+    .shutdownWhenStopped(postStopExecutorService)
+    .build()
+  ```
+- **Session Protocol in RoutingContext**: You can now acquire <type://SessionProtocol> directly from
+  <type://RoutingContext#sessionProtocol()> for enhanced routing and <type://EchangeType> decisions.
+- You can now easily mask request and response headers with ... #5175 #5188
+  - TODO(ikhoon): TBU
+- **Eureka Health Check Customization**: You can override the default health check URL when registering with 
+  Eureka. #5369
+  ```java
+  EurekaUpdatingListener
+    .builder()
+    // Override the default health check path provided by HealthCheckService.
+    .healthCheckUrlPath("/healthcheck")
+    .build();
+  ```
+
+## 📈 Improvements
+
+- Annotated services now support <type://HttpResult> wrapped in RxJava objects #5380 #5386
+- You can now create <type://HttpFile> with 
+  [Spring Boot nested JAR URL](https://docs.spring.io/spring-boot/docs/current/reference/html/executable-jar.html#appendix.executable-jar.jarfile-class.compatibility). #5390 #5391
+- <type://WebsocketService> is restructured as an interface, allowing for custom implementations. #5368
+- Introduce <type://PreTlsHandshakeException> to better report exceptions occurring before a TLS handshake
+  initiates. #5344
+- Error messages for gRPC client stub generation have been clarified. #5387
+
+## 🛠️ Bug fixes
+
+- <type://ClientRequestContext#whenResponseCancelling()> and <type://ClientRequestContext#whenResponseCancelled()>
+  is now correctly completed even if the  `CancellationScheduler` has not been initialized #5212
+- Fixed a leak within a WebSocket client when an invalid response is received. #5412
+- `json_name` attribute in protobuf files is now correctly used for matching query parameters during
+  gRPC HTTP/JSON transcoding. #5193 #5366
+- Implicit request objects are now correctly converted. #5389 #5393
+- Fixed an issue with handling an empty reason phase in WebSocket Close frames. #5388
+- <type://Flags> logging is now consistent, irrespective of <type://RequestContextExportingAppender> 
```

Comment:
Is this generated by the ChatGPT? 😆
I'm not sure what this means. Could you use the original sentence in the commit description?

Reply:
Right... 😆 Many of them were abandoned while polishing the notes.